api/src/paths/activity/{activityId}.ts

```
@@ -1,144 +1,103 @@
+    const db = new QueryHandler({ maintain: true });
 
     try {
-      connection = await getDBConnection();
+      const activityId = req.params.activityId;
       const sqlStatement: SQLStatement = getActivitySqlWithPermissions(activityId, req?.authContext?.user?.user_id);
       const sqlStatement2: SQLStatement = getActivityHistorySQL(activityId);
 
-      if (!sqlStatement || !sqlStatement2) {
-        return res.status(500).json({
-          message: 'Unable to generate SQL statement.',
-          request: req.body,
-          namespace: 'activity/{activityId}',
-          code: 500
-        });
-      }
+      if (!sqlStatement || !sqlStatement2) return res.status(500).send('Unable to generate SQL statement');
 
-      defaultLog.debug({ label: '{activityId}', message: 'activity audit sql ', body: sqlStatement2 });
-      const response1 = await connection.query(sqlStatement.text, sqlStatement.values);
-      const response2 = await connection.query(sqlStatement2.text, sqlStatement2.values);
+      logger.debug('Activity With Permissions SQL', { body: sqlStatement });
+      logger.debug('Activity Audit SQL', { body: sqlStatement2 });
+
+      const response1 = await db.query(sqlStatement);
+      const response2 = await db.query(sqlStatement2);
 
       const result1 = response1?.rows?.[0] ?? null;
       const result2 = response2?.rows ?? null;
 
-      defaultLog.debug({ label: '{activityId}', message: 'activity response', body: JSON.stringify(result1) });
       req['activity'] = result1;
       req['activity_history'] = result2;
-    } catch (error) {
-      defaultLog.debug({ label: 'getActivity', message: 'error', error });
-      return res.status(500).json({
-        message: 'Unable to fetch activity.',
-        request: req.body,
-        error: error,
-        namespace: 'activity/{activityId}',
-        code: 500
-      });
+    } catch (e) {
+      logger.error(e);
+      return res.status(500).send('Unable to fetch activity.');
```

What happens if we omit this and let the catch-all handler take it? Is there any real advantage to this catch block?

---

Removing this and throwing an arbitrary error as a test successfully catches it in the global handler.